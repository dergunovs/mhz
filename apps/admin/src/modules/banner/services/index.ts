import { ComputedRef, Ref } from 'vue';

import { api, useQuery, useMutation, IPageQuery, convertParams } from 'mhz-helpers';
import { API_BANNER, IBanner, IBaseReply, IFilterData } from 'mhz-contracts';

export function getBanners(query: Ref<IPageQuery | number>) {
  async function fn() {
    const params = convertParams(query);

    const { data } = await api.get<{ data: IBanner[]; total: number; filters: IFilterData }>(API_BANNER, { params });

    return data;
  }

  return useQuery({ queryKey: [API_BANNER, query], queryFn: fn });
}

export function getBanner(id?: ComputedRef<string | string[]>) {
  async function fn() {
    if (!id?.value) return null;

    const { data } = await api.get<{ data: IBanner }>(`${API_BANNER}/${id.value}`);

    return data.data;
  }

  return useQuery({ queryKey: [API_BANNER, id], queryFn: fn });
}

export function postBanner(options: object) {
  async function fn(formData: IBanner) {
    const { data } = await api.post<IBaseReply>(API_BANNER, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}

export function updateBanner(options: object) {
  async function fn(formData: IBanner) {
    const { data } = await api.patch<IBaseReply>(`${API_BANNER}/${formData._id}`, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}

export function deleteBanner(options: object) {
  async function fn(id?: string) {
    if (!id) return null;

    const { data } = await api.delete<IBaseReply>(`${API_BANNER}/${id}`);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}
