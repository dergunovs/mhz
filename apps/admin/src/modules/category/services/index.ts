import { ComputedRef } from 'vue';

import { api, useQuery, useMutation } from 'mhz-helpers';
import { API_CATEGORY, ICategory, IBaseReply } from 'mhz-contracts';

export function getCategories(options?: object) {
  async function fn() {
    const { data } = await api.get<{ data: ICategory[] }>(API_CATEGORY);

    return data.data;
  }

  return useQuery({ queryKey: [API_CATEGORY], queryFn: fn, ...options });
}

export function getCategory(id?: ComputedRef<string | string[]>) {
  async function fn() {
    if (!id?.value) return null;

    const { data } = await api.get<{ data: ICategory }>(`${API_CATEGORY}/${id.value}`);

    return data.data;
  }

  return useQuery({ queryKey: [API_CATEGORY, id], queryFn: fn });
}

export function postCategory(options: object) {
  async function fn(formData: ICategory) {
    const { data } = await api.post<IBaseReply>(API_CATEGORY, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}

export function updateCategory(options: object) {
  async function fn(formData: ICategory) {
    const { data } = await api.patch<IBaseReply>(`${API_CATEGORY}/${formData._id}`, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}

export function deleteCategory(options: object) {
  async function fn(id?: string) {
    if (!id) return null;

    const { data } = await api.delete<IBaseReply>(`${API_CATEGORY}/${id}`);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}
