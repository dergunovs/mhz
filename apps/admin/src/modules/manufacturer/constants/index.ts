export const URL_MANUFACTURER = '/manufacturers';
