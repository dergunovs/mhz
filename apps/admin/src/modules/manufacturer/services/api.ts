import { ComputedRef, Ref } from 'vue';

import { api, IPageQuery, convertParams } from 'mhz-helpers';
import { API_MANUFACTURER, IBaseReply, IManufacturer } from 'mhz-contracts';

export async function getManufacturersApi(query: Ref<IPageQuery | number>) {
  const params = convertParams(query);

  const { data } = await api.get<{ data: IManufacturer[]; total: number }>(API_MANUFACTURER, { params });

  return data;
}

export async function getManufacturerApi(id?: ComputedRef<string | string[]>) {
  if (!id?.value) return null;

  const { data } = await api.get<{ data: IManufacturer }>(`${API_MANUFACTURER}/${id.value}`);

  return data.data;
}

export async function postManufacturerApi(formData: IManufacturer) {
  const { data } = await api.post<IBaseReply>(API_MANUFACTURER, formData);

  return data;
}

export async function updateManufacturerApi(formData: IManufacturer) {
  const { data } = await api.patch<IBaseReply>(`${API_MANUFACTURER}/${formData._id}`, formData);

  return data;
}

export async function deleteManufacturerApi(id?: string) {
  if (!id) return null;

  const { data } = await api.delete<IBaseReply>(`${API_MANUFACTURER}/${id}`);

  return data;
}
