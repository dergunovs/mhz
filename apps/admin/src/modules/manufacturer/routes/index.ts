import { URL_MANUFACTURER, URL_MANUFACTURER_CREATE, URL_MANUFACTURER_EDIT } from '@/manufacturer/constants';

export const manufacturerRoutes = [
  {
    path: URL_MANUFACTURER,
    name: 'ManufacturerList',
    component: () => import('@/manufacturer/pages/ManufacturerListPage.vue'),
  },
  {
    path: URL_MANUFACTURER_CREATE,
    name: 'ManufacturerCreate',
    component: () => import('@/manufacturer/pages/ManufacturerCreatePage.vue'),
  },
  {
    path: `${URL_MANUFACTURER_EDIT}/:manufacturer`,
    name: 'ManufacturerEdit',
    component: () => import('@/manufacturer/pages/ManufacturerEditPage.vue'),
  },
];
