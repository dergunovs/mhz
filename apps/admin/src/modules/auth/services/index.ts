import { Ref } from 'vue';
import { useMutation, useQuery } from '@tanstack/vue-query';

import { ILoginFormData } from '@/auth/interface';
import { API_GET_CHECK_AUTH, API_POST_LOGIN } from '@/auth/constants';

import { api } from '@/common/services/api';

export function postLogin(formData: Ref<ILoginFormData>, options: object) {
  async function fn(): Promise<string | undefined> {
    const { data } = await api.post(API_POST_LOGIN, formData.value);

    return data.token;
  }

  return useMutation({
    mutationKey: [API_POST_LOGIN, formData],
    mutationFn: fn,
    ...options,
  });
}

export function getCheckAuth(options: object) {
  async function fn() {
    const { data } = await api.get(API_GET_CHECK_AUTH);

    return data;
  }

  return useQuery({
    queryKey: [API_GET_CHECK_AUTH],
    queryFn: fn,
    ...options,
  });
}
