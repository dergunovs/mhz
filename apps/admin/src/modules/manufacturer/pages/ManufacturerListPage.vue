<template>
  <div>
    <PageTitle>Manufacturers</PageTitle>

    <div :class="$style.page">
      <RouterLink :to="URL_MANUFACTURER_CREATE">Add manufacturer</RouterLink>

      <ManufacturerList :manufacturers="manufacturers" />

      <UiPagination v-if="manufacturers?.length" :page="page" :total="total" @update="updatePage" />
    </div>
  </div>
</template>

<script setup lang="ts">
import { UiPagination } from 'mhz-ui';
import { usePagination, usePage } from 'mhz-helpers';

import PageTitle from '@/layout/components/PageTitle.vue';
import ManufacturerList from '@/manufacturer/components/ManufacturerList.vue';

import { getManufacturers } from '@/manufacturer/services';
import { URL_MANUFACTURER, URL_MANUFACTURER_CREATE } from '@/manufacturer/constants';

const { page } = usePage(URL_MANUFACTURER);

const { data } = getManufacturers(page);

const { data: manufacturers, total, setPage } = usePagination(data);

function updatePage(pageToSet: number) {
  page.value = setPage(pageToSet, page.value);
}
</script>

<style module lang="scss">
.page {
  display: flex;
  flex-direction: column;
  gap: 32px;
}
</style>
