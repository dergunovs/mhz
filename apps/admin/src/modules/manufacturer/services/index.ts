import { useMutation, useQuery } from '@tanstack/vue-query';

import { IManufacturer } from 'mhz-types';

import { API_MANUFACTURER } from '@/manufacturer/constants';
import { api } from '@/common/services/api';

export function getManufacturers() {
  async function fn(): Promise<IManufacturer[] | undefined> {
    const { data } = await api.get(API_MANUFACTURER);

    return data;
  }

  return useQuery({
    queryKey: [API_MANUFACTURER],
    queryFn: fn,
  });
}

export function postManufacturer(options: object) {
  async function fn(formData: IManufacturer) {
    await api.post(API_MANUFACTURER, formData);
  }

  return useMutation({
    mutationKey: [API_MANUFACTURER],
    mutationFn: fn,
    ...options,
  });
}

export function deleteManufacturer(options: object) {
  async function fn(id?: number) {
    await api.delete(`${API_MANUFACTURER}/${id}`);
  }

  return useMutation({
    mutationKey: [API_MANUFACTURER],
    mutationFn: fn,
    ...options,
  });
}
