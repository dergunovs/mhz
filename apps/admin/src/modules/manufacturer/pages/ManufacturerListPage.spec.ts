import { DefineComponent } from 'vue';
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { IManufacturer } from 'mhz-contracts';
import { dataTest } from 'mhz-helpers';

import ManufacturerListPage from './ManufacturerListPage.vue';

import { mockQueryReply, wrapperFactory } from '@/common/test';
import * as manufacturerServices from '@/manufacturer/services';
import { MANUFACTURERS } from '@/manufacturer/fixtures';

const spyGetManufacturers = vi
  .spyOn(manufacturerServices, 'getManufacturers')
  .mockReturnValue(mockQueryReply(MANUFACTURERS));

const manufacturerList = dataTest('manufacturer-list-page-list');

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(ManufacturerListPage, {});
});

enableAutoUnmount(afterEach);

describe('ManufacturerListPage', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(ManufacturerListPage)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('gets manufacturers and passes them to list props', async () => {
    expect(spyGetManufacturers).toBeCalledTimes(1);

    expect(
      wrapper.findComponent<DefineComponent<{ manufacturers: IManufacturer[] }>>(manufacturerList).vm.$props
        .manufacturers
    ).toEqual(MANUFACTURERS.data);
  });
});
