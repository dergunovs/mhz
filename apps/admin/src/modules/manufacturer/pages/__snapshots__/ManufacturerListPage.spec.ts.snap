// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ManufacturerListPage > matches snapshot 1`] = `
"<div>
  <page-title-stub links="[object Object],[object Object]"></page-title-stub>
  <div class="_page_7dde22"><a to="/manufacturers/create">Add manufacturer</a>
    <manufacturer-list-stub manufacturers="[object Object],[object Object],[object Object]" modelvalue="[object Object]" data-test="manufacturer-list-page-list"></manufacturer-list-stub>
    <ui-pagination-stub page="1" total="1" lang="en"></ui-pagination-stub>
  </div>
</div>"
`;
